Sources/Fullscreen/AddressView/AddressView.swift

```diff
@@ -0,0 +1,222 @@
+        view.delegate = self
+        return view
+    }()
+
+    private lazy var addressCardView: AddressCardView = {
+        let view = AddressCardView()
+        view.delegate = self
+        view.translatesAutoresizingMaskIntoConstraints = false
+        return view
+    }()
+
+    private lazy var centerMapButton: UIButton = {
+        let button = UIButton(withAutoLayout: true)
+        button.backgroundColor = .milk
+        button.tintColor = .primaryBlue
+
+        button.layer.cornerRadius = 23
+        button.layer.shadowColor = UIColor.black.cgColor
+        button.layer.shadowOffset = CGSize(width: 0, height: 1)
+        button.layer.shadowRadius = 3
+        button.layer.shadowOpacity = 0.5
+
+        button.setImage(UIImage(named: .pin), for: .normal)
+        button.setImage(UIImage(named: .pin), for: .highlighted)
+        button.addTarget(self, action: #selector(centerMapButtonAction), for: .touchUpInside)
+
+        return button
+    }()
+
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    public required init?(coder aDecoder: NSCoder) {
+        fatalError()
+    }
+
+    public var model: AddressViewModel? {
+        didSet {
+            guard let model = model else { return }
+
+            mapTypeSegmentControl.removeAllSegments()
+            for (index, segment) in model.mapTypes.enumerated() {
+                mapTypeSegmentControl.insertSegment(withTitle: segment, at: index, animated: false)
+            }
+            mapTypeSegmentControl.selectedSegmentIndex = model.selectedMapType
+
+            addressCardView.titleLabel.text = model.title
+            addressCardView.subtitleLabel.text = model.subtitle
+            addressCardView.copyButton.setTitle(model.copyButtonTitle, for: .normal)
+            addressCardView.getDirectionsButton.setTitle(model.getDirectionsButtonTitle, for: .normal)
+        }
+    }
+
+    public func centerMap(location: CLLocationCoordinate2D, regionDistance: Double, animated: Bool) {
+        let meters: CLLocationDistance = regionDistance
+        let region = MKCoordinateRegion(center: location, latitudinalMeters: meters, longitudinalMeters: meters)
+        mapView.setRegion(region, animated: animated)
+    }
+
+    public func addRadiusArea(location: CLLocationCoordinate2D, regionDistance: Double) {
+        mapView.removeOverlays(mapView.overlays)
+        let circle = MKCircle(center: location, radius: regionDistance)
+        mapView.addOverlay(circle)
+    }
+
+    public func addAnnotation(location: CLLocationCoordinate2D, title: String) {
+        mapView.removeAnnotations(mapView.annotations)
+        let annotation = AddressAnnotation(title: title, location: location)
+        mapView.addAnnotation(annotation)
+    }
+
+    public func changeMapType(mapType: MKMapType) {
+        mapView.mapType = mapType
+    }
+}
+
+// MARK: - Private methods
+
+private extension AddressView {
+    private func setup() {
```

Comment:
Think you can remove the private modifier, since it's already in a private extension

Reply:
Fixed!